engine/src/main/java/org/pentaho/di/trans/steps/databaselookup/DatabaseLookup.java

```diff
@@ -679,4 +692,183 @@ private void connectDatabase( Database database ) throws KettleDatabaseException
       logDetailed( BaseMessages.getString( PKG, "DatabaseLookup.Log.ConnectedToDatabase" ) );
     }
   }
+
+  
+  @Override
+  public JSONObject doAction(String fieldName, StepMetaInterface stepMetaInterface, TransMeta transMeta,
+                             Trans trans, Map<String, String> queryParamToValues) {
+    JSONObject response = new JSONObject();
+    try {
+      Method actionMethod = DatabaseLookup.class.getDeclaredMethod(fieldName + "Action", Map.class);
+      this.setStepMetaInterface(stepMetaInterface);
+      response = (JSONObject) actionMethod.invoke(this, queryParamToValues);
+    } catch (NoSuchMethodException e) {
+      log.logError("Method not found: " + e.getMessage(), e);
+      response.put("actionStatus", "FAILURE");
```

Comment:
why are the values "Failure" etc are hard coded, please use the ones present in step Interface as used by other steps

Reply:
accomodated.